event-app-ui/src/components/Profile/Profile.jsx

```diff
@@ -0,0 +1,130 @@
+  const [fullname, setFullname] = useState("");
+  const [email, setEmail] = useState("");
+  const [tel, setTel] = useState("");
+  const [isOpen, setIsOpen] = useState(false);
+  const [eventData, setEventData] = useState([]);
+  const [openProfile, setOpenProfile] = useState(false);
+
+  const navigate = useNavigate();
+  // sending an HTTP request to fetch the users profile information
+
+  // includes cookies and and authentication headers in cross-origin request
+  axios.defaults.withCredentials = true;
+  useEffect(() => {
+    async function getProfileInfo() {
+      try {
+        const response = await axios.get(`http://localhost:3000/profile/${id}`);
+
+        const { username: user, fullname, email, tel } = response.data;
+        setUsername(user);
+        setFullname(fullname);
+        setEmail(email);
+        setTel(tel);
+      } catch (err) {
+        navigate("/login");
+      }
+    }
+    getProfileInfo();
+  }, [id, navigate]);
+  //once the logout button is clicked it redirects the person to the login page
+  const handleLogout = async () => {
+    try {
+      await axios.get("http://localhost:3000/logout");
+    } finally {
+      navigate("/login");
+    }
+  };
+  //fetches thte event data and update the eventData state with the received data
+  useEffect(() => {
+    axios.get(`http://localhost:3000/events`).then((res) => {
+      setEventData(res.data);
+    });
+  }, []);
+
+  return (
+    <div>
+      <Navbar />
+      {/* sending setIsOpen as props popup */}
+
+      <button onClick={handleLogout}>Logout</button>
+      <h1>Welcome {username}!</h1>
+      <div className="profile">
+        <div className="profile-container">
+          <h1>User Profile</h1>
+          <p>Username: {username}</p>
+          {openProfile ? (
+            <EditProfile
+              setOpenProfile={setOpenProfile}
+              fullname={fullname}
+              email={email}
+              tel={tel}
+              username={username}
+            />
+          ) : (
+            <div className="profileInfo">
+              <p>Fullname: {fullname}</p>
+              <p>Email: {email}</p>
+              <p>Tel: {tel}</p>
+              <button
+                onClick={() => {
+                  setOpenProfile(true);
+                }}
+              >
+                Edit Profile
+              </button>
+            </div>
+          )}
+        </div>
+        <div className="listOfEvents">
+          <div>
+            <div className="add-btn">
+              <button
+                className="plus-btn"
+                onClick={() => {
+                  setIsOpen(true);
+                }}
+              >
+                +
+              </button>
+            </div>
+            {isOpen ? (
+              <Popup setIsOpen={setIsOpen} username={username} />
+            ) : (
+              <div>
+                {eventData.map((value, key) => {
+                  if (value.username === username) {
```

Comment:
Oh I see where you're filtering on events matching the username. This is however a security concern. If we return all events to the client, the user can just inspect element and see everyone else's events. It is better to filter on SQL in the back end.

Reply:
I'll do that in the next pull request